kubernetes.go

```diff
@@ -103,6 +119,7 @@ type KubernetesNodePoolUpdateRequest struct {
 	Count     *int              `json:"count,omitempty"`
 	Tags      []string          `json:"tags,omitempty"`
 	Labels    map[string]string `json:"labels,omitempty"`
+	Taints    *[]Taint          `json:"taints,omitempty"`
```

Comment:
Why do we need a pointer here? Just curious because I've noticed that the other structs have `Taints []Taint`.

Reply:
Great question. The pointer is needed here so that we can differentiate between taints being omitted on the update (meaning "I do not want to change the taints at all in my update request") and setting the empty (_not_ nil) slice (meaning "I want to clear all taints").

Taints on the other structs are not pointerized because in the respective cases, there is no need to:

- for the create request in `KubernetesNodePoolCreateRequest`, not setting the taints is equal to leaving the taints empty, so no pointer is needed.
- the `KubernetesNodePool` struct is returned from the API and is meant to indicate which taints are set. We want to be able to indicate if DOKS uses have zero or more taints set, so again no pointer is needed.

Hope this helps.